app/controllers/general_shopping_list_controller.rb

```diff
@@ -0,0 +1,27 @@
+class GeneralShoppingListController < ApplicationController
+  def create
+    @recipe = params[:recipe_id]
+    redirect_to general_shopping_list_index_path(recipe: params[:recipe_id])
+  end
+
+  def index
+    @recipe = Recipe.find(params[:recipe])
+    @recipe_foods = @recipe.recipe_foods.includes(:food)
+    @food_items_to_buy = []
+    @recipe_foods.each do |rfood|
+      item = {
+        name: rfood.food.name,
+        quantity: rfood.food.quantity < rfood.quantity ? rfood.quantity - rfood.food.quantity : rfood.quantity,
+        price: rfood.food.price,
+        measurement_unit: rfood.food.measurement_unit
+      }
```

Comment:
- [ ] At the moment, the general shopping list contains food added to the recipe but it should only show the list of food that is missing for all recipes of the logged-in user. Please kindly adjust to match the requirements.
- [ ] It should also count the total food items and total price of the missing food.